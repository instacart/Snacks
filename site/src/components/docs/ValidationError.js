import React, {Fragment} from 'react'
import Playground from '../Playground'

export default () => (

ValidationError is small red styled text that is used for clientside validation errors. ex TextField:

<Playground>
{`
  <div>
    <TextField
      name="email"
      type="email"
      floatingLabelText="Email"
      hintText="Enter your email address"
      validations={{isEmail: null, isLength: {min: 3, max: 15}}}
      validationErrorText="Please enter a valid email"
      defaultValue="foobars.com"
      isValid={false}
      required
    />
  </div>
`}
</Playground>

)
