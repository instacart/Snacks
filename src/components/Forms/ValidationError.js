import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'
import { colors } from '../../styles'

const styles = {
  root: {
    color: `${colors.RED_700}`,
    fontSize: '12px',
    marginTop: '2px',
    marginLeft: '1px',
    opacity: '0',
    display: 'none',
    pointerEvents: 'none',
    textAlign: 'left',
    transition: 'opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
  },
  show: {
    opacity: '1',
    display: 'initial',
  },
}

@Radium
class ValidationError extends Component {
  static propTypes = {
    /** A uniq id */
    inputId: PropTypes.string.isRequired,
    /** Error text */
    text: PropTypes.string,
    /** Whether to show the error or not */
    show: PropTypes.bool,
  }

  render() {
    const { inputId, show, text } = this.props

    return (
      <div
        style={[styles.root, show && styles.show]}
        aria-live={'assertive'}
        aria-atomic
        id={`error_${inputId}`}
      >
        {text}
      </div>
    )
  }
}

export default ValidationError
