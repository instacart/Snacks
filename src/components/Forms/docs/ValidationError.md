ValidationError is small red styled text that is used for clientside validation errors. ex TextField:

```jsx
import { TextField } from 'ic-snacks'

; <div>
    <TextField
      name="email"
      type="email"
      floatingLabelText="Email"
      hintText="Enter your email address"
      validations={{isEmail: null, isLength: {min: 3, max: 15}}}
      validationErrorText="Please enter a valid email"
      defaultValue="foobars.com"
      isValid={false}
      required
    />
  </div>
```
