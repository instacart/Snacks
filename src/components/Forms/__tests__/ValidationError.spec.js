import React       from 'react'
import renderer    from 'react-test-renderer'
import { StyleRoot } from 'radium'
import ValidationError from '../ValidationError'

it('renders ValidationError correctly', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <ValidationError text="Error Text" show />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders ValidationError correctly when show is false', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <ValidationError text="Error Text" show={false} />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
