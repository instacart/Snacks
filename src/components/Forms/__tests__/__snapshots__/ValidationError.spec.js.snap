// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ValidationError correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      aria-atomic={true}
      aria-live="assertive"
      data-radium={true}
      id="error_foo"
      style={
        Object {
          "color": "#B30029",
          "display": "initial",
          "fontSize": "12px",
          "marginLeft": "1px",
          "marginTop": "2px",
          "opacity": "1",
          "pointerEvents": "none",
          "textAlign": "left",
          "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      Error Text
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`renders ValidationError correctly when show is false 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      aria-atomic={true}
      aria-live="assertive"
      data-radium={true}
      id="error_foo"
      style={
        Object {
          "color": "#B30029",
          "display": "none",
          "fontSize": "12px",
          "marginLeft": "1px",
          "marginTop": "2px",
          "opacity": "0",
          "pointerEvents": "none",
          "textAlign": "left",
          "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      Error Text
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
